test/integration_test.py

```diff
@@ -1857,3 +1860,51 @@ def test_version(self):
             self.assertEqual(response.status_code, 200)
             lambda_status = response.json()['git']
             self.assertEqual(local_status, lambda_status)
+
+
+class ResponseHeadersTest(AzulTestCase):
+
+    def test_response_headers(self):
+
+        root = '/'
+        static = '/static/swagger-ui.css'
+        openapi = '/openapi'
+        oauth = '/oauth2_redirect'
+        health = '/health/basic'
+        global_headers = {
+            'Strict-Transport-Security': 'max-age=31536000; includeSubDomains',
+            'X-Content-Type-Options': 'nosniff',
+            'X-Frame-Options': 'DENY',
+        }
+        test_cases = defaultdict(dict)
+        for endpoint in (root, static, openapi, oauth, health):
+            test_cases[endpoint] |= global_headers
+        for endpoint in (root, oauth):
+            test_cases[endpoint] |= {'Content-Type': 'text/html'}
+        for endpoint in (openapi, health):
+            test_cases[endpoint] |= {'Content-Type': 'application/json'}
+        test_cases[static] |= {'Content-Type': 'text/css'}
+        for endpoint in (root, static, openapi, health):
+            test_cases[endpoint] |= {'Access-Control-Allow-Origin': '*'}
+            test_cases[endpoint] |= {
+                'Access-Control-Allow-Headers': 'Authorization,Content-Type,X-Amz-Date,'
+                                                'X-Amz-Security-Token,X-Api-Key'
+            }
+        for endpoint in (oauth, health):
+            test_cases[endpoint] |= {'Cache-Control': 'no-store'}
+        test_cases[root] |= {'Cache-Control': 'public, max-age=0, must-revalidate'}
+        test_cases[static] |= {
+            'Cache-Control': 'public, max-age=86400'
+        }
+        test_cases[openapi] |= {'Cache-Control': 'public, max-age=500'}
```

Comment:
This looks like it's more code than before. Can you think of something less verbose? Instead of keeping the structure of `test_cases` and assembling it differently, consider inverting the structure (swap keys and values) and going back to a literal. If you can't come up with something, we need to resort to spot checking.